pkg/model/manifest_test.go

```diff
@@ -1297,3 +1297,78 @@ func Test_Manifest_HasBuildSection(t *testing.T) {
 
 	}
 }
+
+func Test_getInferredManifestFromK8sManifestFile(t *testing.T) {
+	wd := t.TempDir()
+	fullpath := filepath.Join(wd, "k8s.yml")
+	f, err := os.Create(fullpath)
+	assert.NoError(t, err)
+	defer func() {
+		if err := f.Close(); err != nil {
+			t.Fatalf("Error closing file %s: %s", fullpath, err)
+		}
+	}()
+	_, err = GetInferredManifest(wd)
+	assert.NoError(t, err)
+}
+
+func Test_getInferredManifestFromK8sManifestFolder(t *testing.T) {
+	wd := t.TempDir()
+	fullpath := filepath.Join(wd, "manifests")
+	assert.NoError(t, os.MkdirAll(filepath.Dir(fullpath), 0750))
+	f, err := os.Create(fullpath)
+	assert.NoError(t, err)
+	defer func() {
+		if err := f.Close(); err != nil {
+			t.Fatalf("Error closing file %s: %s", fullpath, err)
+		}
+	}()
+
+	_, err = GetInferredManifest(wd)
+	assert.NoError(t, err)
+}
+
+func Test_getInferredManifestFromHelmPath(t *testing.T) {
+	var tests = []struct {
+		name          string
+		filesToCreate []string
+		expected      string
+	}{
+		{
+			name:          "chart folder exists on wd",
+			filesToCreate: []string{filepath.Join("chart", "Chart.yaml")},
+			expected:      "charts",
+		},
+		{
+			name:          "chart folder inside helm folder exists on wd",
+			filesToCreate: []string{filepath.Join("helm", "charts", "Chart.yaml")},
+			expected:      filepath.Join("helm", "charts"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			wd := t.TempDir()
+			for _, fileToCreate := range tt.filesToCreate {
+				fullpath := filepath.Join(wd, fileToCreate)
+				assert.NoError(t, os.MkdirAll(filepath.Dir(fullpath), 0750))
+				f, err := os.Create(fullpath)
+				assert.NoError(t, err)
+				defer func() {
+					if err := f.Close(); err != nil {
+						t.Fatalf("Error closing file %s: %s", fullpath, err)
+					}
+				}()
+			}
+			_, err := GetInferredManifest(wd)
```

Comment:
can we check the manifest is the one we are expecting?

Reply:
I went through the other test cases as well mentioned in `manifest_test.go` such as _Test_getManifestFromFile_ in which no manifest is checked.

Regarding the tst case that I've written above, it's running well and checking if the Inferred manifest files of specific names are present or not.  Let me know your thoughts.